### PR TITLE
go/doltcore/sqle/index: use Map.Get() for point lookups

### PR DESCRIPTION
**Pull request by andy-wm-arthur**

### CODE REVIEW

**Comment by andy-wm-arthur** [comment]
#benchmark

**Comment by andy-wm-arthur** [comment]
#benchmark

**Comment by andy-wm-arthur** [comment]
#benchmark

**Comment by max-hoffman** [comment]
The merge benchmarks are ~30% slower with this change, presumably because this reverts one of the optimizations james made to speedup child fk lookups.

**Comment by max-hoffman** [comment]
#systab-benchmark

File: go/libraries/doltcore/merge/violations_fk_prolly.go [hunk 1]
```diff
@@ -169,31 +168,14 @@ func prollyChildSecDiffFkConstraintViolations(
 				}
 			}
 
-			if parentSecIdxCur == nil {
```
Comment (max-hoffman) [inline go/libraries/doltcore/merge/violations_fk_prolly.go:172]: Hit to merge perf is kind of painful. Can we hide this logic in a function in prolly to maintain the private cursor and get the rest of changes through?

**Review by max-hoffman — APPROVED** [review]
LGTM, my only concern is the merge perf regression on child fk lookups

**Comment by andy-wm-arthur** [comment]
#merge-benchmark

File: go/libraries/doltcore/merge/violations_fk_prolly.go (hunk 1 above)
Comment (andy-wm-arthur) [reply go/libraries/doltcore/merge/violations_fk_prolly.go:172]: addressed this [098d40a](https://github.com/dolthub/dolt/pull/5533/commits/098d40a44f669dee5508b7a9309d30c686934da4) with `prolly.Map.HasPrefix()`

**Pull request merged**